lib/tools.js

```diff
@@ -275,14 +275,31 @@ function final (logger, handler) {
   if (typeof logger === 'undefined' || typeof logger.child !== 'function') {
     throw Error('expected a pino logger instance')
   }
-  if (typeof handler !== 'function') {
+  const hasHandler = (typeof handler !== 'undefined')
+  if (hasHandler && typeof handler !== 'function') {
```

Comment:
can we update the error message: "if supplied, the handler parameter should be a function"